modules/ObjectControl/src/objectcontrol.cpp

```diff
@@ -690,6 +714,91 @@ void ObjectControl::startScenario() {
 	}
 }
 
+void ObjectControl::resetTest() {
+	if (this->isResetting) {
+		RCLCPP_INFO(get_logger(), "Test already resetting");
+		return;
+	}
+	this->isResetting = true;
+	RCLCPP_INFO(get_logger(), "Resetting test by offering return trajectories");
+	for (auto& id : getVehicleIDs()) {
+		this->setObjectTrajectory(id);
+	}
+}
+
+void ObjectControl::reloadScenarioTrajectories() {
+	if (!this->isResetting) {
+		RCLCPP_INFO(get_logger(), "Scenario trajectories already loaded");
+		return;
+	}
+	this->isResetting = false;
+	RCLCPP_INFO(get_logger(), "Reloading scenario trajectories");
+	for (auto& id : getVehicleIDs()) {
+		this->setObjectTrajectory(id);
+	}
+}
+
+void ObjectControl::updateTrajectoryGUI(uint32_t id){
+	ATOS::Trajectory traj = objects.at(id)->getTrajectory();
+	this->pathPublishers.emplace(id, ROSChannels::Path::Pub(*this, id));
+	this->pathPublishers.at(id).publish(traj.toPath());
```

Comment:
This doesn't seem to work correctly, when I press `Abort` -> `Clear Abort` -> `Reset Test`, the back to start trajectory shows up for only a short time in the 3D-view, but then the original trajectory is shown again. For the map view however, it's working correctly.

Reply:
I see, I think that `TrajectoryletStreamer` will continue publishing the original trajectory. I can add this as a bug, because `TrajectoryletStreamer` should also publish the back to start trajectory.